godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,818 @@
+                type_name::<T>()
+            )
+        };
+        let name = transfer_string_name_from_godot(p_name);
+
+        let result = handle_panic(ctx, || {
+            let instance = instance_data_as_script_instance::<T>(p_instance);
+
+            borrow_instance(instance).get_property_type(name.clone())
+        });
+
+        let Some(result) = result else {
+            *r_is_valid = transfer_bool_to_godot(false);
+            return 0;
+        };
+
+        *r_is_valid = transfer_bool_to_godot(true);
+        result.sys()
+    }
+
+    /// # Safety
+    ///
+    /// - The engine has to provide a valid pointer to the instance data.
+    /// - `r_is_valid` is expected to be a valid [`GDEXtensionBool`] ptr which can be modified to reflect the validity of the return value.
+    /// - `r_str` is expected to be a string pointer into which the return value can be moved.
+    pub(super) unsafe extern "C" fn to_string_func<T: ScriptInstance>(
+        p_instance: sys::GDExtensionScriptInstanceDataPtr,
+        r_is_valid: *mut sys::GDExtensionBool,
+        r_str: sys::GDExtensionStringPtr,
+    ) {
+        let ctx = || format!("error when calling {}::to_string", type_name::<T>());
+
+        let string = handle_panic(ctx, || {
+            let instance = instance_data_as_script_instance::<T>(p_instance);
+
+            let Ok(inner) = try_borrow_instance(instance) else {
+                // to_string of a script instance can be called when calling to_string on the owning base object. In this case we pretend like we
+                // can't handle the call and leave r_is_valid at it's default value of false.
+                //
+                // This is one of the  only members of GDExtensionScripInstanceInfo which appeares to be called from an API function
+                // (beside get_func, set_func, call_func). The unexpected behavior here is that it is being called as a replacement of Godots
+                // Object::to_string for the owner object. This then also happens when trying to call to_string on the base object inside a
+                // script, which feels wrong, and most importantly, would obviously cause a panic when acquiring the Ref guard.
+
+                return None;
+            };
+
+            Some(inner.to_string())
+        })
+        .flatten();
+
+        let Some(string) = string else {
+            return;
+        };
+
+        *r_is_valid = transfer_bool_to_godot(true);
+        transfer_string_to_godot(string, r_str);
+    }
+
+    /// # Safety    
+    ///
+    /// - The engine has to provide a valid pointer to the instance data.
+    /// - A ptr to a Godot Object reference is returned. The caller is then responsible for freeing it's memory and adjusting the reference
+    ///   count.
+    pub(super) unsafe extern "C" fn get_owner_func<T: ScriptInstance>(
+        p_instance: sys::GDExtensionScriptInstanceDataPtr,
+    ) -> sys::GDExtensionObjectPtr {
```

Comment:
I was not able to find a single usage site for `script_instance->get_owner()` or `script_instance()->get_owner()` in Godot. @Bromeon should we even implement it if it's entirely unused?

Reply:
Removed the function and added a comment to the GDExtension struct.